src/sentry/rules/history/preview.py

```diff
@@ -74,3 +77,48 @@ def preview(
             group_last_fire[event.group_id] = event.timestamp
 
     return Group.objects.filter(id__in=group_ids)
+
+
+def get_events(project: Project, condition_activity: Sequence[ConditionActivity]) -> Dict[str, Any]:
+    """
+    Returns events that have caused issue state changes.
+    """
+    group_ids = []
+    event_ids = []
+    for act in condition_activity:
+        if act.type == ConditionActivityType.CREATE_ISSUE:
+            group_ids.append(act.group_id)
```

Comment:
is there a way we can cut down on some of the complexity here? like we have a list of group_ids, and then we go and get all their events, then we iterate through all of those events and then iterate through all of the activities again

perhaps not but happy to discuss more

Reply:
maybe if we sorted the events by timestamp we wouldn't have to create `group_map`, since the events and condition activity should be in the same relative order. But, we would have to sort them explicitly since they're ordered by `group_id` out of the query